service_capacity_modeling/models/org/netflix/kafka.py

```diff
@@ -100,39 +104,34 @@ def _estimate_kafka_requirement(  # pylint: disable=too-many-positional-argument
         )
     )
 
+    bw_in = (
+        (write_mib_per_second * MIB_IN_BYTES) * copies_per_region
+    ) / MEGABIT_IN_BYTES
+    bw_out = (
+        (
+            (read_mib_per_second * MIB_IN_BYTES)
+            + ((write_mib_per_second * MIB_IN_BYTES) * (copies_per_region - 1))
+        )
+    ) / MEGABIT_IN_BYTES
     if (
-        current_zonal_cluster
-        and current_zonal_cluster.cluster_instance
+        current_zonal_capacity
+        and current_zonal_capacity.cluster_instance
         and required_zone_size is not None
+        and desires.current_clusters is not None
     ):
-        # For now, use the highest average CPU utilization seen on the cluster
-        cpu_utilization = current_zonal_cluster.cpu_utilization.high
-        # Validate with data if we should instead estimate 99th percentile here to get
-        # rid of spikes in collected cpu usage data ?
-        # Use the formula: 99th Percentile ≈ Average + (Z-score * SD).
-        # https://en.wikipedia.org/wiki/Normal_curve_equivalent#:~:text=The%2099th%20percentile%20in%20a,49/2.3263%20=%2021.06.
-        # curr_cpu_util = cpu_util.mid + (2.33 * (cpu_util.high - cpu_util.low) / 6)
-        current_utilized_cores = (
-            current_zonal_cluster.cluster_instance.cpu
-            * required_zone_size
-            * zones_per_region
-            * cpu_utilization
-        ) / 100
-
-        # compute needed core capacity for cluster so avg cpu utilization for the
-        # cluster stays under threshold for that tier
-        needed_cores = int(
-            current_utilized_cores / target_cpu_utilization(desires.service_tier)
+        capacity_requirement = zonal_requirements_from_current(
+            current_cluster=desires.current_clusters,
+            buffers=desires.buffers,
+            instance=current_zonal_capacity.cluster_instance,
+            reference_shape=desires.reference_shape,
         )
-        logger.debug("kafka needed cores: %s", needed_cores)
-        # Normalize those cores to the target shape
-        reference_shape = current_zonal_cluster.cluster_instance
-        needed_cores = normalize_cores(
-            core_count=needed_cores,
-            target_shape=instance,
-            reference_shape=reference_shape,
+        needed_cores = int(capacity_requirement.cpu_cores.mid)
```

Comment:
just using mid cpu cores for needed cores felt not very accurate, hence I had the logic that you have removed. What is rationale for using the cpu.mid here instead?